docs/DeveloperGuide.md

```diff
@@ -1,34 +1,431 @@
+* *SAM* - Short for Stupendously AwsoMe, an acronym to describe the records module.
+* *BRANDON* - Short for BRillant Ahead of its time Neat Dainty OrigNal, it is an acronym to describe the custom 
+implementation of the save file structure. 
+* *A&D* - Short of Amazing & Dazzling, it is an acronym to describe the command module.
+
+###### [Back to top](#table-of-content)
+
+## 2. Design & Implementation
+### 2.1 Project overview
+HAMS is built using java. Under java's write once run anywhere, HAMS can be built on any platform including Windows,
+MAC-OS and Linux. When running locally on these system, HAMS uses as custom save configuration called 
+BRillant Ahead of its time Neat Dainty OrigiNal (BRANDON) for loading and saving data. This allows HAMS' internal 
+list to be modified by operators before and during run time. 
+
+### 2.2 Module Overview
+The major code can be broken down into modules. The below table is the breakdown of the module's custom
+name and a summarized purpose.
+
+**TODO**
+|Module name|Purpose|
+|---------|-------|
+|Stupendously AwesoMe (SAM) **records**|?| 
+|BRillant Ahead of its time Neat Dainty OrigiNal (BRANDON) **storage**|?|
+|Amazing & Dazzling (A&D) Commands|Facade classes that deals with input so that differenct classes can interact with each other|
+|Parser|Parses the user input for command execution|
+
+#### 2.2.1 SAM record module
+
+###### [Back to top](#table-of-content)
+
+#### 2.2.2 BRANDON storage module
+
+###### [Back to top](#table-of-content)
+
+#### 2.2.3 A&D command module 
+
+The command module consist of 11 different classes, where each class does a different command by itself. 
+These classes allows the patients and appointments to be added into HAMS, allows the updating of patient and 
+appointment details, allows the admin to delete patients and appointment information and allows of listing of the 
+different patients and the appointments in HAMS. In addition, the classes also deals with displaying the list of 
+commands and as well as to allowing HAMS to exit.
+
+![](images/Command%20class%20diagram.png)
+
+All of these command classes inherits from the abstract ```Command``` class. Likewise, the execute function of each 
+command  class is also inherited from the abstract ```Command``` class's ```execute(Ui ui, Storage storage)``` function 
+too. Every command class (other than the  ```ExitCommand``` and ```HelpCommand```) are actually façade classes that 
+creates the connection from the ```Main``` class to the other classes required such as ```Storage``` class,
+```PatientList``` class and the ```Appointment``` class to name a few. 
+
+##### 2.2.3.1 AddPatientClass
+
+To add a patient, the ```AddPatientCommand``` class is used. For this ```AddPatientCommand``` class, it serves as the 
+façade class for the ```Main```, ```Patient``` , ```PatientList``` and the ```Storage``` class to interact with one 
+another. 
+
+![](images/AddPatientDiagram.png)
```

Comment:
This diagram is also very complicated. The overlapping arrows are hard to follow.